src/datasources/alarms/components/editors/list-alarms/ListAlarmsQueryEditor.test.tsx

```diff
@@ -347,5 +347,87 @@ describe('ListAlarmsQueryEditor', () => {
         );
       });
     });
+
+    [TransitionInclusionOption.All, TransitionInclusionOption.MostRecentOnly].forEach((option) => {
+      it(`should show transition properties in dropdown when transition inclusion is ${option}`, async () => {
+        const container = await renderElement({
+          refId: 'A',
+          queryType: QueryType.ListAlarms,
+          transitionInclusionOption: option,
+        });
+
+        const propertiesCombobox = container.getAllByRole('combobox')[0];
+        await userEvent.click(propertiesCombobox);
+        await userEvent.clear(propertiesCombobox);
+        await userEvent.type(propertiesCombobox, 'Transition value');
+
+        expect(screen.queryByText('Transition value')).toBeInTheDocument();
+      });
+    });
+
+    it('should not show transition properties in dropdown when transition inclusion is None', async () => {
+      const container = await renderElement({
+        refId: 'A',
+        queryType: QueryType.ListAlarms,
+        transitionInclusionOption: TransitionInclusionOption.None,
+      });
+
+      const propertiesCombobox = container.getAllByRole('combobox')[0];
+      await userEvent.click(propertiesCombobox);
+
+      await userEvent.clear(propertiesCombobox);
+      await userEvent.type(propertiesCombobox, 'Transition condition');
+      expect(screen.queryByText('Transition condition')).not.toBeInTheDocument();
```

Comment:
Follow the "arrange -> act -> assert" pattern as you did in the previous test.


```suggestion
      const container = await renderElement({
        refId: 'A',
        queryType: QueryType.ListAlarms,
        transitionInclusionOption: TransitionInclusionOption.None,
      });

      const propertiesCombobox = container.getAllByRole('combobox')[0];
      await userEvent.click(propertiesCombobox);
      await userEvent.clear(propertiesCombobox);
      await userEvent.type(propertiesCombobox, 'Transition condition');

      expect(screen.queryByText('Transition condition')).not.toBeInTheDocument();
```